modules/ROOT/pages/traversal-framework/index.adoc

```diff
@@ -0,0 +1,57 @@
+:description: The Neo4j Traversal Framework Java API.
+
+:org-neo4j-graphdb-Direction-both: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Direction.html#BOTH
+
+
+[[traversal]]
+= The Traversal Framework
+
+This section provides an overview of the Traversal Framework, and a detailed description of the Neo4j Traversal Framework Java API and how to use it.
+
+The Neo4j Traversal Framework Java API is a callback-based, lazily-executed way of specifying desired movements through a graph in Java.
+Some traversal examples are collected under xref:java-embedded/traversal.adoc[].
+
+Although the Traversal Framework is less readable and more complex than the xref:5.0@cypher-manual:ROOT:index.adoc[Cypher query language], it
+offers a powerful approach to traversing the graph. Being able to dynamically make custom choices at each step
```

Comment:
Not sure if "code" is the right word here... but I was thinking something like "process", "program", "calculation"?

```suggestion
The ability to dynamically make custom choices at each step of the traversal makes the code more expressive and potentially more performant than Cypher.
```

Reply:
Agreed that this section could be improved :+1: